artman/pipelines/grpc_generation.py

```diff
@@ -25,156 +25,84 @@ class GrpcClientPipeline(code_gen.CodeGenerationPipelineBase):
 
     def __init__(self, **kwargs):
         super(GrpcClientPipeline, self).__init__(
-            get_grpc_task_factory(kwargs), **kwargs)
+            ProtoGenTaskFactory(gen_grpc=True, **kwargs), **kwargs)
 
 
 class ProtoClientPipeline(code_gen.CodeGenerationPipelineBase):
 
     def __init__(self, **kwargs):
         super(ProtoClientPipeline, self).__init__(
-            get_proto_task_factory(kwargs), **kwargs)
+            ProtoGenTaskFactory(gen_grpc=False, **kwargs), **kwargs)
 
 
-class GrpcTaskFactoryBase(code_gen.TaskFactoryBase):
+class ProtoGenTaskFactory(code_gen.TaskFactoryBase):
+
+    def __init__(self, gen_grpc, **kwargs):
+        if 'language' not in kwargs:
+            raise ValueError('Valid `language` argument required for gRPC codegen')
+        if 'aspect' not in kwargs:
+            raise ValueError('Valid `aspect` argument required for gRPC codegen')
+        self.language = kwargs['language']
+        self.gen_grpc = gen_grpc
+        self.gen_code = kwargs['aspect'] == 'ALL' or kwargs['aspect'] == 'CODE'
+        self.gen_pkg = kwargs['aspect'] == 'ALL' or kwargs['aspect'] == 'PACKAGE'
 
     def get_tasks(self, **kwargs):
         tasks = self.get_grpc_codegen_tasks(**kwargs)
         tasks += self._get_publish_tasks(**kwargs)
         return task_utils.instantiate_tasks(tasks, kwargs)
 
     def get_grpc_codegen_tasks(self, **kwargs):
+        if self.language == 'java':
```

Comment:
There's not really much that is shared, and there's not really much logic either, it's basically just providing a sequence of commands. I feel like trying to share stuff here creates more abstraction complexity than is worth it. I'm trying to cut out abstractions and make this as concrete as possible.

Reply:
Can you at least put them in separate methods (each `if` is a method)? maybe even then you can use map instead of if-else statements (i'm not sure, but worth trying)

Reply:
Can you at least put them in separate methods (each `if` is a method)? maybe even then you can use map instead of if-else statements (i'm not sure, but worth trying)

Reply:
Ok I got a map to methods to work.